hoot-core/src/main/cpp/hoot/core/conflate/DiffConflator.cpp

```diff
@@ -130,65 +132,102 @@ void DiffConflator::apply(OsmMapPtr& map)
   _stats.append(SingleStat("Project to Planar Time (sec)", _timer.getElapsedAndRestart()));
   OsmMapWriterFactory::writeDebugMap(_map, "after-projecting-to-planar");
 
-  // find all the matches in this map
+  // Find all the matches in this map.
   _intraDatasetMatchOnlyElementIds.clear();
   _intraDatasetElementIdsPopulated = false;
   _createMatches();
+  if (ConfigOptions().getWriterIncludeConflateScoreTags())
+  {
+    // Add score tags to all matches that have some score component.
+    _addConflateScoreTags();
+  }
 
   _currentStep++;
 
-  if (_conflateTags)
-  {
-    // Use matches to calculate and store tag diff. We must do this before we create the map diff,
-    // because that operation deletes all of the info needed for calculating the tag diff.
-    _updateProgress(currentStep - 1, "Storing tag differentials...");
-    _calcAndStoreTagChanges();
-    _currentStep++;
-  }
+  if (!ConfigOptions().getConflateMatchOnly())
```

Comment:
The only way I could get the partial match removal to work was if the matches were optimized like they are with Unifying. However when doing completely removal only, I would get strange output when non-linear matches got optimized. So, only optimizing linear matches and only doing so if a linear matcher is in the config.